bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/Capability.java

```diff
@@ -55,44 +58,42 @@ public class Capability {
     Capability(CommonInterface handler) {
         this.handler = handler;
         this.thing = handler.getThing();
+        this.moduleType = ModuleType.from(thing.getThingTypeUID());
     }
 
-    public final @Nullable String setNewData(@Nullable NAObject newData) {
-        if (newData != null) {
-            beforeNewData();
-            if (newData instanceof HomeData) {
-                updateHomeData((HomeData) newData);
-            }
-            if (newData instanceof HomeStatus) {
-                updateHomeStatus((HomeStatus) newData);
-            }
-            if (newData instanceof HomeStatusModule) {
-                updateHomeStatusModule((HomeStatusModule) newData);
-            }
-            if (newData instanceof Event) {
-                updateEvent((Event) newData);
-            }
-            if (newData instanceof HomeEvent) {
-                updateHomeEvent((HomeEvent) newData);
-            }
-            if (newData instanceof NAThing) {
-                updateNAThing((NAThing) newData);
-            }
-            if (newData instanceof NAMain) {
-                updateNAMain((NAMain) newData);
-            }
-            if (newData instanceof Device) {
-                updateNADevice((Device) newData);
-            }
-            afterNewData(newData);
+    public final @Nullable String setNewData(NAObject newData) {
+        beforeNewData();
+        if (newData instanceof HomeData) {
+            updateHomeData((HomeData) newData);
+        }
```

Comment:
change all 
```
}

if (
```
to
```
} else if (
```

Reply:
No, some are exclusive, some not. It is by design. Each channelHelper does its part of the job.